### PR TITLE
CDAP-2283: separate MetricStore TagValue from Cube API

### PR DESCRIPTION
**Pull request by abaranau**
Two changes:

* created MetricTagValue in proto, to isolate HTTP APIs from internally used classes
* created TagValue in watchdog-api to separate from one that is used in Cube API
The TagValue in Cube API will be renamed to DimValue, i.e. dimension value

Update: added commit that renames tag to dimension in Cube: https://github.com/caskdata/cdap/commit/dbb2514957b9f365fcb15b4de4667548f6d05913

### CODE REVIEW

File: cdap-proto/src/main/java/co/cask/cdap/proto/MetricTagValue.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  @Override
+  public int hashCode() {
+    return name.hashCode() + (value == null ? 0 : 31 * value.hashCode());
+  }
+
+  @Override
+  public String toString() {
+    final StringBuilder sb = new StringBuilder();
+    sb.append("TagValue");
```
Comment (gsps1) [inline cdap-proto/src/main/java/co/cask/cdap/proto/MetricTagValue.java]: MetricTagValue

File: cdap-proto/src/main/java/co/cask/cdap/proto/MetricTagValue.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ */
+
+package co.cask.cdap.proto;
+
+import co.cask.cdap.api.dataset.lib.cube.CubeFact;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents tag and its value associated with {@link CubeFact}.
```
Comment (gsps1) [inline cdap-proto/src/main/java/co/cask/cdap/proto/MetricTagValue.java]: not CubeFact right ? should be MetricsHandler

File: cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/TagValue.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+package co.cask.cdap.api.metrics;
+
+import co.cask.cdap.api.annotation.Beta;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents tag and its value associated with {@link MetricValues}.
+ */
```
Comment (gsps1) [inline cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/TagValue.java]: not sure if its MetricValues

**Comment by gsps1** [comment]
just more question. LGTM :+1:

File: cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/TagValue.java (hunk 3 above)
Comment (abaranau) [inline cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/TagValue.java]: it is

**Comment by abaranau** [comment]
http://builds.cask.co/browse/CDAP-DUT1674-3 GREEN

**Pull request merged**